spring-integration-rsocket/src/main/java/org/springframework/integration/rsocket/outbound/RSocketOutboundGateway.java

```diff
@@ -112,10 +108,25 @@ public RSocketOutboundGateway(String route, @Nullable Object... routeVariables)
 	 * in this expression evaluation, for example using some bean with an appropriate logic.
 	 * @param routeExpression the SpEL expression to use.
 	 */
-	@SuppressWarnings("this-escape")
 	public RSocketOutboundGateway(Expression routeExpression) {
+		this(routeExpression, null);
+	}
+
+	/**
+	 * Instantiate based on the provided SpEL expression to evaluate an RSocket endpoint {@code route}
+	 * at runtime against a request message.
+	 * If route is a template and variables expansion is required, it is recommended to do that
+	 * in this expression evaluation, for example using some bean with an appropriate logic.
+	 * @param routeExpression the SpEL expression to use.
+	 * @param routeVariables the variables to expand route template.
+	 */
+	@SuppressWarnings("this-escape")
+	public RSocketOutboundGateway(Expression routeExpression, Object @Nullable [] routeVariables) {
```

Comment:
Please, revise this to be a `private` ctor.
I don't want to encourage users to follow this pattern.
It is really better to do that how it is explained in JavaDocs: fully via expression.
The expression is going to be evaluated against request message at runtime, and there you can build the route whatever way you need.